misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -296,6 +297,13 @@ ClientConfigHolder getClientConfigHolder() {
                     .filter(JBallerinaDebugServer::isValidFrame)
                     .toArray(StackFrame[]::new);
 
+            if (context.getLastInstruction() == DebugInstruction.STEP_IN
+                && validFrames[0].getSource().getPath().equals(topStackFrame.getSource().getPath())
+                && validFrames[0].getLine().intValue() == topStackFrame.getLine()) {
+                eventProcessor.sendStepRequest(args.getThreadId(), StepRequest.STEP_INTO);
+                context.setLastInstruction(null);
+            }
+            topStackFrame = validFrames[0];
```

Comment:
Potential `IndexOutOfBoundsException` as the `validFrames` can be an empty array

Reply:
Fixed